### DISCUSSION TITLE
Can the PC send data to a HID device ?

### DISCUSSION THREAD

**Discussion by redo33** [hathach/tinyusb#2909] [Q&A]
Hi,

I wonder whether there is a way to send data to a USB HID device from the PC.  

It seems to do it using a USB device custom HID class.

Any advice would be greatly appreciated.

Thanks.

**Comment by hathach** [comment]
It is rather simple, check out the hid_generic_inout example

**Reply by redo33** [reply]
Thank you very much.   Finally [it](https://github.com/hathach/tinyusb/tree/master/examples/device/hid_generic_inout) works !

Windows 11

```
d:\dev\python hid_inout.py

VID list: cafe, 239a, 2e8a, 303a
{'path': b'\\\\?\\HID#VID_303A&PID_80C2&MI_00#c&39af95b4&1&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}', 'vendor_id': 12346, 'product_id': 32962, 'serial_number': '0', 'release_number': 256, 'manufacturer_string': 'WEMOS.CC', 'product_string': 'TinyUSB HID', 'usage_page': 1, 'usage': 8, 'interface_number': 0, 'bus_type': <BusType.USB: 1>}
Send text to HID Device :
```

I also created a code snippet for testing below:

```
import hid

# Wemos ESP32 S2
vid = 0x303a
pid = 0x80c2

with hid.Device(vid, pid) as h:
	print(f'Device manufacturer: {h.manufacturer}')
	print(f'Product: {h.product}')
	print(f'Serial Number: {h.serial}') 
```

**Comment by cr1901** [comment]
Just adding to what @hathach says:

Yes, it can be done. You're limited to 64 kB per second (64 bytes, 1000 times per second), but there are many times this is enough.
